Assets/Scripts/Models/Job/RequestedItem.cs

```diff
@@ -0,0 +1,84 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using MoonSharp.Interpreter;
+using UnityEngine;
+
+namespace ProjectPorcupine.Jobs
+{
+    [MoonSharpUserData]
+    public class RequestedItem
+    {
+        public RequestedItem(string type, int minAmountRequested, int maxAmountRequested)
+        {
+            Type = type;
+            MinAmountRequested = minAmountRequested;
+            MaxAmountRequested = maxAmountRequested;
+        }
+
+        public RequestedItem(string type, int maxAmountRequested)
+            : this(type, maxAmountRequested, maxAmountRequested)
+        {
+        }
+
+        public RequestedItem(RequestedItem item)
+            : this(item.Type, item.MinAmountRequested, item.MaxAmountRequested)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProjectPorcupine.Job.RequestedItem"/> class with the amount needed to fill the inventory item passed in.
+        /// </summary>
+        public RequestedItem(Inventory inventory)
+            : this(inventory.Type, inventory.MaxStackSize - inventory.StackSize)
+        {
+        }
+
+        public string Type { get; set; }
+
+        public int MinAmountRequested { get; set; }
+
+        public int MaxAmountRequested { get; set; }
+
+        public RequestedItem Clone()
+        {
+            return new RequestedItem(this);
+        }
+
+        public bool NeedsMore(Inventory inventory)
+        {
+            return AmountNeeded(inventory) > 0;
+        }
+
+        public bool DesiresMore(Inventory inventory)
+        {
+            return AmountDesired(inventory) > 0;
+        }
+
+        public int AmountDesired(Inventory inventory = null)
+        {
+            if (inventory == null || inventory.Type != Type)
+            {
+                return MaxAmountRequested;
```

Comment:
I think that the idea behind min and max will be that a Metal Smelter can say: "Hey, I want 5 Raw Iron to work but you bring me upto 50 since I will use it eventually". The Character would pick a job, and ask: "How many items do you need?", and the answer will be 5 Raw Iron. He goes to look for 5 Raw Iron and finds a stack of 20. So he can then ask: "Are you cool with 20 Raw Iron", and the Job would say, "Yes, he can hold upto 50". So the Character replies: "Cool, I'll take al 20 and we can save on trips later on". Not sure if this was implemented.

Saying that I think that Desires more is a bit confusing, yes it can take more, but it really desires the min amount to start working. Maybe something like CanTakeMore or CanHoldMore, and the other function can be MaxAmountItCanHold or something like that.